include/lsst/partition/MapReduce.h

```diff
@@ -536,7 +537,7 @@ namespace detail {
         std::vector<SortedRecordRange> ranges;
         ranges.reserve(_numWorkers);
         // BEWARE: a lock on _mutex is held while constructing Worker.
-        // This is because _vm contains strings and the current GCC libstc++
+        // This is because _config contains strings and the current GCC libstc++
```

Comment:
Is this whole comment still relevant? Maybe it's enough to say "because ConfigStore is not thread safe"?

Reply:
I don't fully understand yet the `MapReduce` implementation to remove the lock on the mutex. The new config system certainly has strings in its implementation. Let's just keep the lock for now until the next major refactoring of the package.